src/system/Form/InputWithCopyButton.js

```diff
@@ -0,0 +1,107 @@
+/** @jsxImportSource theme-ui */
+
+/**
+ * External dependencies
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import { MdContentCopy } from 'react-icons/md';
+
+/**
+ * Internal dependencies
+ */
+import { Button, Validation, Label } from '..';
+import { Input as ThemeInput } from 'theme-ui';
+import { baseControlStyle } from './Input.styles';
+
+const inputStyles = {
+	unset: 'all',
+	...baseControlStyle,
+	lineHeight: 'inherit',
+	minHeight: '36px',
+	px: 3,
+	py: 2,
+	fontSize: 2,
+	mb: 2,
+	variant: 'inputs.default',
+};
+
+const InputWithCopyButton = React.forwardRef(
+	(
+		{ variant, label, forLabel, hasError = false, required, sx = {}, errorMessage, ...props },
+		ref
+	) => {
+		if ( ! ref ) {
+			ref = React.createRef();
+		}
+
+		const handleCopy = e => {
+			e.preventDefault();
+			if ( ! ref.current ) {
+				return;
+			}
+			const originalTypeStatus = ref.current.getAttribute( 'type' ) || '';
+			if ( ref.current.getAttribute( 'value' ) === '' ) {
+				ref.current.setAttribute( 'value', '    ' );
+			}
+			ref.current.setAttribute( 'type', 'text' );
+			ref.current.select();
+			document.execCommand( 'copy' ); // eslint-disable-line no-undef
```

Comment:
Looks like `execCommand` is [deprecated](https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand#browser_compatibility), should we use the [Clipboard API](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API) instead, to ensure it's future-proof?

Reply:
We could also use React to retrieve the value as I suspect this will be mostly for read-only operations. But that's not a blocker for the PR, just an idea ;)

Reply:
I will try this!

Reply:
Resolved in https://github.com/Automattic/vip-design-system/pull/206/commits/7fca9708bf9ee4e91841d7c8dedf378f04f921d8